File: android/app/src/main/java/net/melisma/selko/data/model/EventChangeProposal.kt
package net.melisma.selko.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonObject
import kotlin.time.Instant

@Serializable
enum class EventChangeProposalKind {
    @SerialName("material_update") MATERIAL_UPDATE,
    CANCELLATION
}

@Serializable
enum class EventChangeProposalStatus {
    @SerialName("pending") PENDING,
    @SerialName("applied") APPLIED,
    @SerialName("rejected") REJECTED,
    @SerialName("superseded") SUPERSEDED,
    @SerialName("closed_legacy") CLOSED_LEGACY
}

@Serializable
data class EventChangeProposal(
    val id: String,
    @SerialName("event_id") val eventId: String,
    @SerialName("user_id") val userId: String,
    @SerialName("source_id") val sourceId: String,
    val kind: EventChangeProposalKind,
    val status: EventChangeProposalStatus,
    @SerialName("change_set") val changeSet: EventChangeSet,
    @SerialName("event_snapshot_before") val eventSnapshotBefore: JsonObject? = null,
    @SerialName("resolution_reason") val resolutionReason: String? = null,
    @SerialName("created_at") val createdAt: Instant? = null,
    @SerialName("resolved_at") val resolvedAt: Instant? = null,
    @SerialName("updated_at") val updatedAt: Instant? = null
)

File: ios/Selko/Features/Events/Models/EventChangeProposal.swift
import Foundation

enum EventChangeProposalKind: String, Codable, Sendable {
    case materialUpdate = "material_update"
    case cancellation
}

enum EventChangeProposalStatus: String, Codable, Sendable {
    case pending
    case applied
    case rejected
    case superseded
    case closedLegacy = "closed_legacy"
}

struct EventChangeProposal: Identifiable, Codable, Sendable, Equatable {
    let id: UUID
    let eventId: UUID
    let userId: UUID
    let sourceId: UUID
    let kind: EventChangeProposalKind
    let status: EventChangeProposalStatus
    let changeSet: EventChangeSet
    let resolutionReason: String?
    let createdAt: Date?
    let resolvedAt: Date?
    let updatedAt: Date?

    enum CodingKeys: String, CodingKey {
        case id
        case eventId = "event_id"
        case userId = "user_id"
        case sourceId = "source_id"
        case kind, status
        case changeSet = "change_set"
        case resolutionReason = "resolution_reason"
        case createdAt = "created_at"
        case resolvedAt = "resolved_at"
        case updatedAt = "updated_at"
    }
}
